File: xssfinder-php-executor/test/XssFinder/Runner/LabelledXssGeneratorImplTest.php
<?php

namespace XssFinder\Runner;

use PHPUnit_Framework_TestCase;
use XssFinder\Annotations\LabelledXssGenerator;
use XssFinder\Xss\XssAttack;
use XssFinder\Xss\XssGenerator;

class LabelledXssGeneratorImplTest extends PHPUnit_Framework_TestCase
{
    const LABEL = 'label';
    const XSS_ATTACK_STRING = 'some attack';
    const XSS_ATTACK_ID = '123';

    /** @var XssGenerator */
    private $_mockXssGenerator;
    /** @var XssAttack */
    private $_mockXssAttack;

    /** @var LabelledXssGeneratorImpl */
    private $_labelledXssGenerator;

    function setUp()
    {
        $this->_mockXssGenerator = mock('XssFinder\Xss\XssGenerator');
        $this->_mockXssAttack = mock('XssFinder\Xss\XssAttack');
        when($this->_mockXssGenerator->createXssAttack())->return($this->_mockXssAttack);
        when($this->_mockXssAttack->getIdentifier())->return(self::XSS_ATTACK_ID);
        when($this->_mockXssAttack->getAttackString())->return(self::XSS_ATTACK_STRING);

        $this->_labelledXssGenerator = new LabelledXssGeneratorImpl($this->_mockXssGenerator);
    }

    function testXssIsGenerated()
    {
        // when
        $attackString = $this->_labelledXssGenerator->getXssAttackTextForLabel(self::LABEL);

        // then
        assertThat($attackString, is(self::XSS_ATTACK_STRING));
    }

    function testGeneratedXssIdIsAvailableKeyedByLabel()
    {
        // given
        $this->_labelledXssGenerator->getXssAttackTextForLabel(self::LABEL);

        // when
        $labelToXssId = $this->_labelledXssGenerator->getLabelsToAttackIds();

        // then
        $expectedLabelToXssId = array(self::LABEL => self::XSS_ATTACK_ID);
        assertThat($labelToXssId, is($expectedLabelToXssId));
    }
}

File: xssfinder-java-executor/src/test/java/org/xssfinder/runner/LabelledXssGeneratorImplTest.java
package org.xssfinder.runner;

import com.google.common.collect.ImmutableMap;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.runners.MockitoJUnitRunner;
import org.xssfinder.xss.XssAttack;
import org.xssfinder.xss.XssGenerator;

import java.util.Map;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;
import static org.mockito.Mockito.when;

@RunWith(MockitoJUnitRunner.class)
public class LabelledXssGeneratorImplTest {
    private static final String LABEL = "someLabel";
    private static final String XSS_TEXT = "some XSS";
    private static final String ATTACK_ID = "attackId";

    @Mock
    private XssGenerator mockXssGenerator;
    @Mock
    private XssAttack mockAttack;

    @InjectMocks
    private LabelledXssGeneratorImpl labelledXssGenerator;

    @Before
    public void setUp() {
        when(mockAttack.getAttackString()).thenReturn(XSS_TEXT);
        when(mockAttack.getIdentifier()).thenReturn(ATTACK_ID);
        when(mockXssGenerator.createXssAttack()).thenReturn(mockAttack);
    }

    @Test
    public void xssIsGenerated() {
        // when
        String xss = labelledXssGenerator.getXssAttackTextForLabel(LABEL);

        // then
        assertThat(xss, is(XSS_TEXT));
    }

    @Test
    public void generatedXssIdIsAvailableKeyedByLabel() {
        // given
        labelledXssGenerator.getXssAttackTextForLabel(LABEL);

        // when
        Map<String,String> labelToXssId = labelledXssGenerator.getLabelsToAttackIds();

        // then
        Map<String, String> expectedLabelToXssId = ImmutableMap.of(LABEL, ATTACK_ID);
        assertThat(labelToXssId, is(expectedLabelToXssId));
    }
}
